[REF] l10n_*_website_sale: refactor E-commerce localizations

kcv-odoo opened the pull request.

robodoo commented:
This PR targets the un-managed branch odoo-dev/odoo:master-portal-allow-multi-address-kcv, it needs to be retargeted before it can be merged.

kcv-odoo commented on addons/l10n_ar_website_sale/demo/website_demo.xml:
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo noupdate="1">
-
-    <record id="default_website_ri" model="website">
```
Check with accounting team if it's okay to remove this and bellow product.pricelist demo data

kcv-odoo commented on addons/l10n_br_website_sale/tests/test_address.py:
```diff
@@ -6,8 +6,7 @@
 class TestBrazilianAddress(HttpCase):
     def test_brazilian_address_frontend(self):
```
Move these tours in as test module or somewhere or make tour for portal or both

kcv-odoo commented on addons/l10n_ar_website_sale/demo/website_demo.xml:
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo noupdate="1">
-
-    <record id="default_website_ri" model="website">
-        <field name="name">(AR) Responsable Inscripto Website</field>
-        <field name="company_id" ref="base.company_ri"/>
-        <field name="domain" model="ir.config_parameter" eval="obj().env['ir.config_parameter'].sudo().get_param('web.base.url')"/>
-    </record>
```
Is it okay to remove this demo data? As we need bridge module just to change website name and company
Same for price-list to set website field on it

kcv-odoo commented on addons/l10n_test_website_sale/static/tests/tours/l10n_br_checkout_address.js:
Move tours related to accounting localization and shop address into a test module l10n_test_website_sale to ensure shop address is not broken after this refactoring and be able to remove those module which only contains tours after refactoring

kcv-odoo commented on addons/l10n_ec_website_sale/demo/website_demo.xml:
```diff
@@ -1,17 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo noupdate="1">
-
-    <record id="default_website_ec" model="website">
```
same as AR

kcv-odoo reviewed (commented):
Hello @jco-odoo :wave: 

Here I have removed website_sale related localisations and add move them in main module so I can also be used in Portal addresses
I have to remove some demo data where they change website name and other things and keep some modules which contains additional E-com related code and created l10n_test_website_sale module to move related tour to ensure I am not breaking anything :see_no_evil: 

Can you take a quick look at PR if there is any bigger issue with this approach?
(so I can move it in my main branch)

Thank you :blush:

robodoo commented:
This PR targets the un-managed branch odoo-dev/odoo:master-portal-allow-multi-address-kcv, it needs to be retargeted before it can be merged.

jco-odoo commented:
@kcv-odoo I don't think there is a big problem with removing the demo data.  It was also not consistently there.  t is nice to just have this common test module as well.  It would be good that @vbe-odoo @masi-odoo @fvz-odoo test the main  master-portal-allow-multi-address-localizations-kcv further.

masi-odoo commented:
Hello @jco-odoo thanks for advising on this E-commerce refactor.

@kcv-odoo on my side, testing the eCommerce for Chile I had no issues and LGTM. (Although I still need to install the l10n_cl_edi_website_sale module).

Regarding the eCommerce for Colombia module I still needed to install (l10n_co_edi_website_sale) and I do have one small comment:

1. When selecting NIT as the `Identification Type`, it is necessary to add a `Company Name`, when not adding any data to this field and clicking on the "Continue to checkout" button the page keeps loading but not continuing to the next step:

![image](https://github.com/user-attachments/assets/69b32ec2-9dce-4768-91d4-6ff821ee3ed9)

Is it possible you could take a look? A part from that it worked correctly. Thanks!

fvz-odoo commented:
Hello @kcv-odoo 

I was checking for l10n_pe and I see some things:

- First, I don't see the module anymore (although I do see the fields for type of identification and the ones related to the customer's address, which were added in the module)
Your branch:
![Screenshot 2025-01-02 at 2 42 25 PM](https://github.com/user-attachments/assets/d5dd8d1a-6971-4c19-a690-36c891eb8026)
Regular Runbot:
![Screenshot 2025-01-02 at 2 43 47 PM](https://github.com/user-attachments/assets/770dbff7-9e55-451f-bf3b-687623866a60)

- Second, when I select any type of identification, I get a traceback error (but after i hit close in the error it let's me go through)
![Screenshot 2025-01-02 at 2 04 52 PM](https://github.com/user-attachments/assets/2aa612c4-607f-49bd-8dbb-1b9215e3c14d)

Other than that it is working as it should. 😉

Let me know if removing the module was intentionally part of the PR and if you can take a look at the traceback.

Cheers!

Fernando

@jco-odoo

kcv-odoo commented:
Hello @jco-odoo Great!! :partying_face: 

Hello @masi-odoo @fvz-odoo 

FYI I moved this code in main branch `master-portal-allow-multi-address-kcv` so you can test further in those branch
Also you can test
![image](https://github.com/user-attachments/assets/d3754bb8-af15-4399-96ce-21c368d91b1b)
Also we are adding possibility to allow users to add multiple address even when E-commerce is not install so user can manage their address from portal

> When selecting NIT as the Identification Type, it is necessary to add a Company Name, when not adding any data to this field and clicking on the "Continue to checkout" button the page keeps loading but not continuing to the next step:

I'll take a look into in and ask for your review in main branch once it done

> First, I don't see the module anymore (although I do see the fields for type of identification and the ones related to the customer's address, which were added in the module)
Your branch:

Yes we removed that module and moved address related code in portal so user can add multiple address even without E-commerce is not installed and this field are there in portal address and user need to set them specific to country according to their localization

> Second, when I select any type of identification, I get a traceback error (but after i hit close in the error it let's me go through)

It does not seems to related to my changes. I'll check if I am able to reproduce and compare with master and try to solve the issue

cc: @Feyensv
